feat(conditions): add :nullAddress as automatically injected context variable

theref opened the pull request:
## Summary

Adds `:nullAddress` as an automatically injected context variable that resolves to the Ethereum null address (0x0000000000000000000000000000000000000000) on the node side.

## Motivation

This is the taco-web companion PR to nucypher/nucypher#3668, which adds the `:nullAddress` protected context variable to the nucypher node implementation.

## Changes

- Add `:nullAddress` to AUTOMATICALLY_INJECTED_CONTEXT_PARAMS array
- Add `:nullAddress` to RESERVED_CONTEXT_PARAMS array to prevent manual setting by users

## Testing

The existing test suite already handles automatically injected parameters:
- Tests verify that automatically injected parameters cannot be manually set
- Tests verify that reserved parameters are properly protected

## Related PRs

- nucypher/nucypher#3668 - Node-side implementation of :nullAddress context variable

theref committed:
feat(conditions): add :nullAddress as automatically injected context variable

This context variable is automatically injected on the node side and resolves
to the Ethereum null address (0x0000000000000000000000000000000000000000).

- Add :nullAddress to AUTOMATICALLY_INJECTED_CONTEXT_PARAMS
- Add :nullAddress to RESERVED_CONTEXT_PARAMS to prevent manual setting
- Existing tests already handle automatically injected parameters

Related to nucypher/nucypher#3668

derekpierre commented on packages/taco/src/conditions/context/context.ts:
```diff
@@ -68,10 +63,12 @@ const EXPECTED_AUTH_PROVIDER_TYPES: Record<string, AuthProviderType[]> = {
 export const AUTOMATICALLY_INJECTED_CONTEXT_PARAMS = [
   // These context parameters are automatically injected on the node side.
   SIGNING_CONDITION_OBJECT_CONTEXT_VAR,
+  ':nullAddress',
```
Let's make this a constant somewhere, and then use the constant in both places.

theref committed:
refactor(conditions): Extract nullAddress string literal into constant

Replace ':nullAddress' string literals with NULL_ADDRESS_CONTEXT_VAR
constant defined in conditions/const.ts, used in both
AUTOMATICALLY_INJECTED_CONTEXT_PARAMS and RESERVED_CONTEXT_PARAMS.

derekpierre reviewed (approved):
🎸
